lib/Cmts.ml

```diff
@@ -247,19 +247,43 @@ let relocate_pattern_matching_cmts (t : t) src tok ~whole_loc ~matched_loc =
     Option.value_exn (Source.loc_of_first_token_at src whole_loc tok)
   in
   let f map =
-    let before, after =
-      List.partition_tf (Map.find_multi map matched_loc)
-        ~f:(fun Cmt.{loc; _} -> Location.compare_end loc kwd_loc < 0)
-    in
-    let map =
-      List.fold_left ~init:map (List.rev before) ~f:(fun map data ->
-          Map.add_multi map ~key:whole_loc ~data )
-    in
-    Multimap.change_multi map matched_loc after
+    Multimap.partition_multi map ~src:matched_loc ~dst:whole_loc
+      ~f:(fun Cmt.{loc; _} -> Location.compare_end loc kwd_loc < 0)
   in
   update_cmts t `Before ~f ;
   update_cmts t `Within ~f
 
+let relocate_ext_cmts (t : t) src ((_pre : string Location.loc), pld)
+    ~whole_loc =
+  match pld with
+  | PStr
+      [ { pstr_desc=
+            Pstr_eval
+              ( { pexp_desc= Pexp_constant (Pconst_string _)
+                ; pexp_loc= _
+                ; pexp_loc_stack= _
+                ; pexp_attributes= _ }
+              , [] )
+        ; pstr_loc } ]
+    when Source.is_quoted_string src pstr_loc ->
+      ()
+  | PStr [{pstr_desc= Pstr_eval _; pstr_loc; _}] ->
+      let kwd_loc =
+        match Source.loc_of_first_token_at src whole_loc LBRACKETPERCENT with
+        | Some loc -> loc
+        | None -> (
+          match Source.loc_of_first_token_at src whole_loc PERCENT with
+          | Some loc -> loc
+          | None -> impossible "expect token starting extension" )
+      in
+      let f map =
+        Multimap.partition_multi map ~src:pstr_loc ~dst:whole_loc
+          ~f:(fun Cmt.{loc; _} -> Location.compare loc kwd_loc < 0)
+      in
+      update_cmts t `Before ~f ;
+      update_cmts t `Within ~f
```

Comment:
This is the same pattern as in `relocate_pattern_matching_cmts`  (partition_multi_map based on loc comparison, and update before and within) - can you pull a function for it?

Reply:
Good idea, done!